### DISCUSSION TITLE
استفاده از دیتابیس SqlLite

### DISCUSSION THREAD

**Discussion by milad-i** [Sina-Soltani/Parbad.Samples#12] [Q&A] [answered]
لطفا نحوه ی استفاده از دیتابیس SqlLite به عنوان storage رو هم آموزش بدید.

سوالی که دارم اینه که با توجه به توضیحات شما که استفاده از پایگاه داده، برای مصرف و عملکرد داخلی پَرباد است، آیا پاک شدن ناگهانی دیتابیس SqlLite از پروژه مشکلی ایجاد میکنه؟

**Comment by Sina-Soltani** [comment]
تفاوت خاصی بین تنظیمات و پیاده سازی sqlite با sqlserver نیست.
أموزش مربوط به sqlserver و همچنین نمونه پروژه رو نگاه کنید و فقط به جای sqlserver از sqlite استفاده کنید.

در مورد سوال دوم:
اگر منظورتون از پاک شدن دیتابیس اینه که یعنی دیتابیس به دلیل اتفاقاتی اصلا وجود نداشته باشه، که خب مسلما مشکلی که به وجود میاد اینه که برنامه موقع ذخیره و بازیابی خطا تولید میکنه چون دیتابیسی وجود نداره که بخواد باهاش کار کنه.

اما اگر منظورتون اینه که اطلاعات فعلی موجود در دیتابیس حذف بشن، در این حالت مشکلی که ممکنه به وجود بیاد اینه که رکورد های فعلی اطلاعات که شامل آیدی ها، تراکنش ها و آخرین وضعیت پرداخت ها دیگه وجود ندارن که بشه ازشون استفاده ای کرد و تصمیماتی گرفت.
برای مثال کاربری در سیستم شما عملیات پرداخت انجام میده و به صفحه درگاه پرداخت میره و در حین پرداخت، اطلاعات موجود در دیتابیس شما بنا بر دلایلی حذف میشه. زمانی که کاربر دوباره به سیستم شما هدایت میشه، عملیات تایید تراکنش باید انجام بشه. اما هیچ اطلاعاتی موجود نیست که بر اساس اون ها تصمیمی گرفته بشه و عملیاتی چک بشه و اطلاعاتی با درگاه بانکی رد و بدل بشه. بنابراین Parbad خطایی صادر میکنه که عملیات نامعتبر هست

**Comment by milad-i** [comment]
زمان کانفیگ کردن Storage، در حال حاظر من از UseSqlServer استفاده میکنم. ولی گزینه UseSqlLite یا مشابه اون نیست. حتی بسته Microsoft.EntityFrameworkCore.Sqlite رو هم نصب کردم ولی بازم نمیدونم چطور میشه از sqllite استفاده کردن.
ممنون میشم نمونه کدی بزارید
![Parbad](https://user-images.githubusercontent.com/55461845/169662165-0357acfe-017e-4681-bbe0-f1bd6710a4ab.png)

**Comment by Sina-Soltani** [comment, marked as answer]
لطفا سوالتون رو در فروم‌های برنامه نویسی جستجو کنید چون این گونه سوال‌ها ارتباطی با این پکیج ندارن.
اگر پکیج Microsoft.EntityFrameworkCore.Sqlite رو نصب کرده باشید، متد UseSqlite هم مسلما قابل مشاهده هست.
اگر این متد رو نمی بینید،‌ یعنی اینکه پکیج رو نصب نکردید و یا اینکه به گونه ای یک اشتباهی در جایی انجام دادید.

آموزش نصب و کار با EntityFrameworkCore و Sqlite رو در سایت آموزشی مایکروسافت مشاهده کنید و بررسی کنید آیا همه چیز رو درست انجام دادید یا خیر.
https://docs.microsoft.com/en-us/ef/core/get-started/overview/first-app?tabs=netcore-cli

پس از نصب پکیج،‌ همونطور که گفتم تنظیمات مربوط به Parbad کاملا با نمونه کد SqlServer یکی هست و کافیه از متد UseSqlite استفاده کنید

```c#
var assemblyName = typeof(Program).Assembly.GetName().Name;

options.ConfigureDbContext = db => db.UseSqlite("ConnectionString", sql => sql.MigrationsAssembly(assemblyName));
```

**Comment by milad-i** [comment]
جناب سلطانی واقعا ممنونم از تمام زحمات و پاسخگویی شما. مطمئنم بقیه کاربرا هم همچین حسی دارن.
 واقعا با حوصله و جزییات کامل توضیح میدید.
 امیدوارم همیشه موفق و پیروز باشید.

**Reply by Sina-Soltani** [reply]
خواهش میکنم.
موفق باشید.
